src/Native/Bootstrap/main.cpp

```diff
@@ -29,13 +44,10 @@ extern "C" void __managedcode_a();
 extern "C" void __managedcode_z();
 
 #else
-// TODO: Once the dotnet compiler is modified to configure the linker to merge sections
-// on OSX / Linux, store these externs in the .modules section and remove work-around
-// in ModuleHeaderNode.cs.
-//extern "C" __attribute((section ("__DATA,.modules$A"))) void * __modules_a[];
-//extern "C" __attribute((section ("__DATA,.modules$Z"))) void * __modules_z[];
-//__attribute((section ("__DATA,.modules$A"))) void * __modules_a[] = { nullptr };
-//__attribute((section ("__DATA,.modules$Z"))) void * __modules_z[] = { nullptr };
+extern "C" __attribute((section (".modules$A"))) void * __modules_a[];
+extern "C" __attribute((section (".modules$Z"))) void * __modules_z[];
+__attribute((section (".modules$A"))) void * __modules_a[] = { nullptr };
+__attribute((section (".modules$Z"))) void * __modules_z[] = { nullptr };
 
 extern "C" void * __modules_a[];
```

Comment:
Not sure yet - I should have tagged this as WIP since I need to get OSX working before this is done.  Jan moved this code into bootstrap which should be the only place it's used so we may no longer need the externs.